internal/config/profile.go

```diff
@@ -225,18 +225,48 @@ func (p *profile) SetOrgID(v string) {
 	p.Set(orgID, v)
 }
 
+// GetConfigDescription returns a map describing the configuration
+func GetConfigDescription() (map[string]string, error) {
+	if err := p.Load(false); err != nil {
+		return nil, err
+	}
```

Comment:
you can avoid reloading the config (loading of the config happens automatically on [cli run](https://github.com/mongodb/mongocli/blob/e9b2e2b338b5577cfbe7231a08a7bd96ef3e2fb1/cmd/mongocli/mongocli.go#L104-L108))

and do `viper.GetStringMapString(profileName)` this will also ignore env variables as it just reads from the file
to ignore api keys then you can do something like
```go
if _, ok := settings[privateAPIKey]; ok {
    settings[privateAPIKey] = "redacted"
}

Reply:
Yeah that will probably be better, as it avoids accidentally changing the config